apps/api/src/app/workflows/usecases/update-workflow-override/update-workflow-override.usecase.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+
+import {
+  NotificationTemplateEntity,
+  NotificationTemplateRepository,
+  TenantEntity,
+  TenantRepository,
+  WorkflowOverrideRepository,
+  WorkflowOverrideEntity,
+} from '@novu/dal';
+import { UpdateWorkflowOverrideCommand } from './update-workflow-override.command';
+import { UpdateWorkflowOverrideResponseDto } from '../../dto/update-workflow-override-response.dto';
+
+@Injectable()
+export class UpdateWorkflowOverride {
+  constructor(
+    private tenantRepository: TenantRepository,
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private workflowOverrideRepository: WorkflowOverrideRepository
+  ) {}
+
+  async execute(command: UpdateWorkflowOverrideCommand): Promise<UpdateWorkflowOverrideResponseDto> {
+    const { tenant, workflow } = await this.extractEntities(command);
+
+    const currentOverrideEntity = await this.workflowOverrideRepository.findOne({
+      _environmentId: command.environmentId,
+      _workflowId: workflow._id,
+      _tenantId: tenant._id,
+    });
```

Comment:
The reason for the refatch is to validate that the entity exists and if it missing alert the user what exactly missing. 
Do you think it could be removed and keep only `Workflow override with workflow id ${command._workflowId} and tenant identifier ${command.tenantIdentifier} was not found` 

Another reason is to help the user by providing him the ability to use the identifiers from the UI dashboard, the same we do with the 'triggerIdentifier'.